WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -499,23 +506,49 @@ private extension ProductsViewController {
                                                  sortOrder: sortOrder)
     }
 
+    /// Configure resultController.
+    /// Assign closures and start performBatch
+    ///
     func configureResultsController(_ resultsController: ResultsController<StorageProduct>, onReload: @escaping () -> Void) {
+        setClosuresToResultController(resultsController, onReload: onReload)
+
+        do {
+            try resultsController.performFetch()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
+
+        tableView.reloadData()
+    }
+
+    /// Set closure  to methods `onDidChangeContent` and `onDidResetContent
+    ///
+    func setClosuresToResultController(_ resultsController: ResultsController<StorageProduct>, onReload: @escaping () -> Void) {
         resultsController.onDidChangeContent = {
             onReload()
         }
 
         resultsController.onDidResetContent = {
             onReload()
         }
+    }
 
-        do {
-            try resultsController.performFetch()
-        } catch {
-            ServiceLocator.crashLogging.logError(error)
-        }
-
+    /// Manages view components and reload tableview
+    ///
+    func reloadTableAndView() {
+        showOrHideToolBar()
+        removeOverlayIfControllerHasResults()
```

Comment:
I think we could update this function to something along the lines of `updateOverlays` and decide if we need to show or hide the overlay depending on the content of our results controller, because at the moment, we have a case where the "no products" overlay is not shown after we remove the last product. And we are forced to pull to refresh.


https://user-images.githubusercontent.com/562080/124985118-0c8f2500-e008-11eb-8e43-0d95e663ec47.mp4

Reply:
Added! I think this bug happened before this task, so we fixed two different bugs doing this PR, that's awesome! 💪